packages/crayons-core/src/components/icon/icon-fetch-memoize.js

```diff
@@ -0,0 +1,39 @@
+import { CRAYONS_ICONS_ASSET_PATH } from '@freshworks/crayons-icon';
+
+const iconCache = {};
+const requestCache = {};
+
+export async function fetchIcon(icon, lib, lib_path) {
+  const icon_key = `${lib}-${icon}`;
+
+  if (iconCache[icon_key]) {
+    return iconCache[icon_key];
+  }
+
+  lib === 'custom'
+    ? (lib_path = `${window.location.origin}/${lib_path}`)
```

Comment:
Curious to know in which circumstance we would need to use a relative URL, I mean real-world apps would mostly be publishing to a CDN. 

Also, the library type called "custom" may be confusing to support the use case of relative urls. What if the users map an absolute URL and name the library custom.

Reply:
@detiwari003 Can you check this comment?